### PR TITLE
import work following testing

### PR DESCRIPTION
**Pull request by quadrismegistus**
FIrst pass at solving some thorny importing issues. Most issues should now be resolved.

See https://github.com/Princeton-CDH/mep-django/issues/791 and https://github.com/Princeton-CDH/mep-django/issues/792 for testing notes.

### CODE REVIEW

File: mep/common/admin.py [hunk 1]
```diff
@@ -80,6 +80,26 @@ def before_import_row(self, row, **kwargs):
         """
         pass
 
+    def validate_row_by_slug(self, row):
+        if not row.get("slug"):
```
Comment (rlskoeser) [inline mep/common/admin.py:84]: let's add a doc string to document what this is; here's a draft version with my understanding of what's going on here (may need more detail)

```suggestion
     """Make sure the record to update can be found by slug or past slug; if the slug is a past slug, row data is updated to use the current slug."""
        if not row.get("slug"):
```

File: mep/common/admin.py [hunk 2]
```diff
@@ -80,6 +80,26 @@ def before_import_row(self, row, **kwargs):
...
+            return False
+        if not self.Meta.model.objects.filter(slug=row["slug"]).exists():
+            # past slug?
+            work = self.Meta.model.objects.filter(past_slugs__slug=row["slug"]).first()
+            if work:
+                logger.debug(f'renaming {row["slug"]} to {work.slug}')
```
Comment (rlskoeser) [inline mep/common/admin.py:90]: your variable here is `work` but it looks like you've implemented for both person and works? (which btw is great); you could call it `instance` or `record`

I don't think this log message out of context would make sense to me; here's a suggested revision

```suggestion
                logger.debug(f'Record found by past slug {row["slug"]}, updating to {work.slug}')
```

**Review by rlskoeser — APPROVED** [review]
Nice work on the past slug logic.  This looks good.

I've suggested some wording / docstring additions or changes. I'd like you to at least rename the db migration (but better if you can consolidate), and if you could briefly look and see if you can keep the unique constraint without it complaining about nulls (which I think was the original intention).

Doesn't need to be reviewed again, you can merge when you're satisfied.

File: mep/books/migrations/0028_auto_20240212_1300.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Generated by Django 3.2.23 on 2024-02-12 18:00
```
[the remark this answers is not in the document]
Comment (quadrismegistus) [reply mep/books/migrations/0028_auto_20240212_1300.py:1]: I was wondering about that—how can I do that? I think the develop version has the first migration already

[the remark this answers is not in the document]
Comment (rlskoeser) [reply mep/books/migrations/0028_auto_20240212_1300.py:1]: it's ok if it's in develop, just not in prod.

here's what I would do:
- apply the migration in reverse
- remove the migration
- create new migration with meaningful name
- git rm the old one and git add the new one

**Pull request merged**